packages/hardhat-core/test/internal/hardhat-network/stack-traces/compilation.ts

```diff
@@ -34,14 +34,6 @@ function getSolcInput(
 ): CompilerInput {
   const isViaIR = compilerOptions.optimizer?.viaIR ?? false;
 
-  const optimizerDetails = isViaIR
-    ? {
-        yulDetails: {
-          optimizerSteps: "u",
```

Comment:
So the issue with using `"u"` as `optimizerSteps` is that it does not allow stack-to-memory movement for some reason.

However using default `optimizerSteps` leads to failure of the `console-logs::error::assert`. It seems the reason is that source mappings of viaIR compiled files are a mess and mostly unusable.

@fvictorio if you have any insights on the issue with the source mappings, I'd appreciate that.